### ISSUE TITLE
Improvement of Magento store configuration with separate file

### ISSUE THREAD

**Issue by thomasReflet** [graphcommerce-org/graphcommerce#1422]
### Describe the feature you'd like to request

Provide a more detailed configuration file for stores to deal with M2 store_code, i18n configuration, subfolders and others store's specific configurations.
Allow M2 i18n locale not based on website's subfolder.
Allow different localizations for countries sharing the same language (en-us, en-gb...)

### Describe the solution you'd like

Upgrade .env file to remove store configurations and move them into a separate json file to provide more options by store.
EG.:
```
STORES = {
    'en' : {
        'store_code': 'en_int',
        'i18n': 'en',
        'incl_tax': true,
        ...
    },
    'en-us' : {
        'store_code': 'en_us',
        'i18n': 'en-us',
        'incl_tax': true,
        ...
    },
    'en-gb': {
        'store_code': 'en_gb',
        'i18n': 'en-gb',
        'incl_tax': true,
        ...
    },
    'jp': {
        'store_code': 'ja_jp',
        'i18n': 'ja-jp',
        'incl_tax': false,
        ...
    },
    ...
}
```

This is an exemple with `/subfolder/` as main key but it can also be M2 `store_code`
We can also imagine lingui i18n fallback and plural configuration here, and maybe different base urls by stores in the future

### Describe alternatives you've considered

Add a second configuration in .env file to separate M2 `store_code` locale and i18n locale:

```
NEXT_PUBLIC_LOCALE_STORES='{"en-us": "en_US", "nl": "nl_NL", "fr-be": "fr_BE", "nl-be": "nl_BE", "en-gb": "en_GB", "en-ca": "en_CA"}'

NEXT_PUBLIC_LOCALE_I18N='{"en-us": "en-us", "nl": "nl-nl", "fr-be": "fr-be", "nl-be": "nl-be", "en-gb": "en-gb", "en-ca": "en-ca"}'
```

and provide a `localeToI18n` function in lingui-next to adapt `LinguiProvider.tsx` and `withLingui.tsx`

**Comment by paales** [comment]
Might be interesting to take a look here as well: https://nextjs.org/docs/advanced-features/i18n-routing Maybe we should 'simply' add some additional props to this configuration

**Comment by paales** [comment]
Available in GraphCommerce 6.0.0
